### PR TITLE
Adds missing support for lexicographical comparison for numeric operators

### LINKED ISSUE #3: It appears that putting an "and" condition in a "filter" does not work
Hey, not sure if you're still doing work on this. It's a neat idea, and I was a little disappointed that http://jsonlogic.com doesn't have a C# implementation, so checked yours out.

I ran into an issue with rules that look like this. Oddly enough, if i just remove part 2 of the "and" condition, it works as expected. 
```
{
  "filter": [
    { "var": "" },
    {
      "and": [
        {
          ">=": [
            { "var": "Prop1.PropA" },
           1
          ]
        },
        {
          ">=": [
            { "var": "Prop2.PropB" },
            19
          ]
        }
      ]
    }
  ]
}

```
with data that looks like this

```
[
  {
    "Prop1": {
      "PropA": 5
    },
    "Prop2": {
      "PropB": 18
    }
  },
  {
    "Prop1": {
      "PropA": 1
    },
    "Prop2": {
      "PropB": 35
    }
  }
]
```

Any ideas?

Cheers,
Stephen

### LINKED ISSUE #14: Support for lexicographical comparison for numeric operators
I want to use the < and > operators to compare strings lexicographically (Dates in form of yyyy-MM-dd to be more precisely). This works in JS but unfortunately not in this .Net port. It throws a FormatException because the strings cannot be converted to double.

I forked and tried to fix this but I'm having a hard time doing so due to the `Func` based nature in the code.

As a first try, I have changed `DoubleArgsSatisfy` so that it works in a generic way. But I'm still only able to set one type when calling this method in `AddOperator`

    AddOperator(">", ArgsSatisfy<string>((prev, next) => string.CompareOrdinal(prev, next) > 0));

    private Func<IProcessJsonLogic, JToken[], object, object> ArgsSatisfy<T>(Func<T, T, bool> criteria)
    {
        return (p, args, data) => {
            var valuesString = args.Select(a => a == null ? "" : p.Apply(a, data).ToString()).ToArray();
            var values = valuesString.Select(v => (T) Convert.ChangeType(v, typeof(T))).ToArray();
            for (int i = 1; i < values.Length; i++) {
                if (!criteria(values[i-1], values[i])) return false;
            }
            return true;
        };
    }

Can you give me a hint on how to improve this?

### PR DESCRIPTION
**Pull request by sarensw**
This pull request adds missing support for lexicographical comparison for numeric operators. Operators supported are <, >, <=, >=. 
This is especially useful for comparison of dates, for example.

Fixes #14

### CODE REVIEW

**Comment by sarensw** [comment]
@yavuztor the build is failing with the following error message "Error signing output with 

    public key from file '../../signingKey.snk'

I had the same issue on my local dev machine and resolved it by removing this link in the `AssemblyInfo.cs` file. I'm not sure why this is needed.

**Comment by yavuztor** [comment]
@sarensw , it took me a while to get to this, sorry for that.  When I added the strong naming, I added the private key, which has caused the build failure you have experienced. I pushed the change that fixes the issue.

File: JsonLogic.Net/EvaluateOperators.cs [hunk 1]
```diff
@@ -300,19 +300,37 @@ private Type DictionaryType(object d)
...
+                    .Where(a => a != null)
+                    .Select(a => p.Apply(a, data))
+                    .Select(a => JToken.FromObject(a))
+                    .ToArray();
+                
+                // all values text?
+                var allText = values.All(a => a.Type == JTokenType.String);
+                if (allText)
+                {
+                    var valuesText = args.Select(a => a == null ? "" : p.Apply(a, data).ToString()).ToArray();
```
Comment (yavuztor) [inline JsonLogic.Net/EvaluateOperators.cs]: the values are already generated. There is no need to re-apply the evaluations. You can use the values like `var textValues = values.Cast<strirng>().ToArray();`

File: JsonLogic.Net/EvaluateOperators.cs [hunk 2]
```diff
@@ -300,19 +300,37 @@ private Type DictionaryType(object d)
...
+            {
+                var values = args
+                    .Where(a => a != null)
+                    .Select(a => p.Apply(a, data))
+                    .Select(a => JToken.FromObject(a))
+                    .ToArray();
+                
+                // all values text?
+                var allText = values.All(a => a.Type == JTokenType.String);
+                if (allText)
```
Comment (yavuztor) [inline JsonLogic.Net/EvaluateOperators.cs]: With this change, this method handles two cases: text and double. We can extract each of those cases into a method and then simplify this method to something like:
```
return allText ? CheckCriteria(values.Cast<string>()) : CheckCriteria(values.Cast<double>());
```

**Comment by yavuztor** [comment]
Closing this because of inactivity.

**Pull request closed without merging**